### PR TITLE
Abductor mode

### PR DESCRIPTION
**Pull request by MacHac**
Refer to issue #705.
### Intent of Pull Request

This PR adds two 'new' modes to the game, traitor + abductor and traitor + abductor + changeling.  It also adds them to the secret rotation.  This will allow players to play abductor again, without any of the problems traditionally associated with abductor only rounds.

Feedback encouraged.

### CODE REVIEW

**Comment by Groudonmaster** [comment]
I think we should probably fix the Abductor mechanics i.e. the stupid console being broken atm before we add them back in.

**Comment by X-TheDark** [comment]
Indeed, please do not merge this until the abductor mode itself (machines needing 30 seconds to bring icons back, maybe some glands balancing) is fixed/balanced.

**Comment by Super3222** [comment]
traitorabchan

what what

Also I thought you worked out the balance issues with Adam already?

**Comment by FluffySurvivor** [comment]
this means we will see traitors and lings more often (besides abductor)
:-1:

**Comment by ShadowDeath6** [comment]
**DO NOT HAVE THIS IN CONJUNCTION WITH CHANGELING HOLY FUCK**

:-1: x :100: IF THIS IS WITH CHANGELING

**Comment by MacHac** [comment]
@Super3222 The balance issue fixes are not in the code right now, I wanted to get the PR up first.
@ShadowDeath6 Changelings are one of the abductor's weaknesses, and their possible presence could help ensure that the abductors don't get sloppy.  The probe mode on the stun baton can identify them.
@FluffySurvivor The chances could be switched around.

I'm going to see if I can't figure out what the console is doing.

**Comment by MacHac** [comment]
Console is fixed by the latest commit.  I'll look at the balancing factors.

**Comment by ShadowDeath6** [comment]
> probe mode tells who's changeling

Pretty sure it just tells their race, not their antag status.

Also keep in mind that abductor glands can be pure cancer to literally everyone in the round, and there will be a lot of people's antag rounds ruined by this.

**Comment by MacHac** [comment]
No, it does changelings too.  Changelings have a special alert.

**Comment by MacHac** [comment]
All the no-questions-asked round ruining glands are commented out.  Virus is still in because you can fix its effects by curing the one with the organ.

**Comment by ShadowDeath6** [comment]
> curing the one with the organ

I thought you couldn't catch the disease if you had the virus.

**Comment by X-TheDark** [comment]
Ok, seems latest commits removed some pretty annoying glands and fixed the console buttons (and by extension others, since it's in mob/life rather than tied to the console), I think this should be given a go. I'm all for variety and having EVERYONE GET IN HERE sounds like something fun (to at least try).

:+1:

File: code/game/gamemodes/abduction/traitor_abd.dm [hunk 1]
```diff
@@ -0,0 +1,264 @@
+/datum/game_mode/traitor/abduction
+	name = "traitor+abductor"
+	config_tag = "traitorabd"
+	traitors_possible = 3 //hard limit on traitors if scaling is turned off
+	restricted_jobs = list("AI", "Cyborg")
+	required_players = 0    // Should only spawn traitors during extreme lowpop, but wierd things do happen
```
Comment (X-TheDark) [inline code/game/gamemodes/abduction/traitor_abd.dm]: I'd change this to something like 20 purely to make sure nothing breaks **too** hard.

File: code/game/gamemodes/abduction/traitor_abd_ling.dm [hunk 2]
```diff
@@ -0,0 +1,79 @@
+/datum/game_mode/traitor/abduction/changeling
+	name = "traitor+abductor+changeling" //MADNESS
+	config_tag = "traitorabdchan"
+	traitors_possible = 2 //hard limit on traitors if scaling is turned off
+	restricted_jobs = list("AI", "Cyborg")
+	required_players = 0
```
Comment (X-TheDark) [inline code/game/gamemodes/abduction/traitor_abd_ling.dm]: I'd change this to 20-30, again, to make sure there's some players to go around.

**Comment by MacHac** [comment]
Yeah, at the very least implement the buttons change.  This should stop shadowlings from having this problem too if they hivemind too often.

@ShadowDeath6 They're given the disease, but with the carrier flag on, so they don't manifest any symptoms themselves.  They can still be cured.

File: code/game/gamemodes/abduction/traitor_abd.dm (hunk 1 above)
Comment (MacHac) [inline code/game/gamemodes/abduction/traitor_abd.dm]: Left that in for testing, will change it

File: code/game/gamemodes/abduction/traitor_abd_ling.dm (hunk 2 above)
Comment (MacHac) [inline code/game/gamemodes/abduction/traitor_abd_ling.dm]: I would also like to add that abductor teams can't spawn if there aren't at least 6 people without a special role per team, which would regulate things a bit

**Comment by ShadowDeath6** [comment]
How many more people does traitor + abductor need than regular traitor or traitorling?

**Comment by MacHac** [comment]
Regular traitor and traitorling both don't need any actual non-antags, because they can (theoretically) be greentext'd even if you're the only person on the server.   This is why you will always be either traitor or traitorling if you join an empty server.

**Comment by MacHac** [comment]
Are there any other problems?  Any other reasons why this shouldn't be merged?

**Pull request merged**